pkg/lifecycle/daemon/routes_navcycle_getstep.go

```diff
@@ -26,6 +27,16 @@ func (d *NavcycleRoutes) getStep(c *gin.Context) {
 				return
 			}
 
+			if preExecuteFunc, exists := d.PreExecuteFuncMap[step.ShortName()]; exists {
+				if err := preExecuteFunc(context.Background(), step); err != nil {
```

Comment:
Should we be using background context here? What do these preExecuteFuncs do?

Reply:
OK, looks like your current use of this does not use context yet, but this is still worth considering before we merge.

Reply:
discussed irl - background context from the request handler is passed to pre execute funcs